### ISSUE TITLE
Add accessibilityHint and accessibilityLabel support to ButtonGroup buttons

### ISSUE THREAD

**Issue by mitchdowney** [react-native-elements/react-native-elements#3196]
**Is your feature request related to a problem? Please Describe.**

Add accessibilityHint and accessibilityLabel support to ButtonGroup buttons. Currently when I try to pass in a custom button component like this to the ButtonGroup buttons prop:

```
const buttons = [
  (
    <Text
      accessibilityHint={translate('ARIA HINT - Tap to search by podcast title')}
      accessibilityLabel={translate('Search by podcast title')}>
      {translate('Podcast')}
    </Text>
  ),
  (
    <Text
      accessibilityHint={translate('ARIA HINT - Tap to search by podcast host')}
      accessibilityLabel={translate('Search by podcast title')}>
      {translate('Host')}
    </Text>
  )
]
```

Neither the hint or label are read. I think it may be because the ButtonGroup buttons are wrapped in their own <Touchable> that we cannot override or configure with props.

**Describe the solution you'd like**

When accessibilityHint and accessibilityLabel are provided for custom ButtonGroup buttons, they should be read by iOS and Android screen readers. This solution may include overriding the <Touchable> wrapping the ButtonGroup buttons.

**Describe alternatives you've considered**

I can't think of another way to accomplish the same thing.

**Comment by ruiaraujolink** [comment]
I'm facing the same problem.

**Comment by ruiaraujolink** [comment]
@mitchdowney 
My workaround is this:

```javascript
buttons={[
  <View accessibilityLabel='Ver gráfico' collapsable key='chart'>
    <FontAwesomeIcon
      color={selectedTabIndex === 0 ? theme.colors?.highlight : theme.colors?.primary}
      icon={faChartLine}
      size={20}
    />
  </View>,
  <View accessibilityLabel='Ver tabela' collapsable key='table'>
    <FontAwesomeIcon
      color={selectedTabIndex === 1 ? theme.colors?.highlight : theme.colors?.primary}
      icon={faTable}
      size={20}
    />
  </View>,
]}
```

Make a View collapsable enable me to put the accessibilityLabel and this approach allow me to pass the accessibility Scanner test on Android.

**Comment by mitchdowney** [comment]
@ruiaraujolink oh interesting, thanks for sharing.

Any idea why `collapsable` is needed here for Android only? How the heck did you figure that out? lol

Also, is it OK to keep this property on iOS too? It won't cause any conflicts?

**Comment by ruiaraujolink** [comment]
@mitchdowney Well on the last day I have tried many options to make this works because I'm working on an app for one public institute in Portugal and accessibility is needed. 

After reading all properties of a View component, I decided to try the `collapsable` option, and then the problem was solved :D

I only have tried on Android with the Accessibility Scanner app. 
I have run some simple tests in IOS and have no issue, but not try the IOS accessibility scanner, only the click functions, and this has worked fine. (The release to App Store was delayed, so, I'm focusing on Android for now).

**Comment by kurisubrooks** [comment]
I'm having this problem on iOS and I've tried the above workarounds, but they're not working. 
When using VoiceOver on iOS with the above fixes, nothing is read-out. I've checked with Accessibility Inspector as well and the label is blank. 

The only time its read-out is if I pass in an array of strings (instead of basic `<View accessibilityLabel="string"><Text>String</Text></View>` components), but this isn't sufficient as no "Button" role is attached to the item.

I'd appreciate it if this issue could be a little higher priority as it completely breaks accessibility.

![image](https://user-images.githubusercontent.com/6291467/144530857-618363ef-cdb2-48fa-9cf4-ef6afe08cfb7.png)

**Comment by yndue736** [comment]
This is very impressive and also a useful idea you ca check [little big city 2](https://apkmission.com/little-big-city-2-mod-apk/) where you can apply this kind of ideas and it will be a suitable choice for tis project.

**Comment by yndue736** [comment]
This is very amazing you can check here some information about [chilli comms](https://www.chillicomms.com/) where you can get the right information about the issues of my problem.

**Comment by inbuilt-aura** [comment]
hey, @adrianPasalega,can I contribute to this project?plzz assign me

**Comment by arpitBhalla** [comment]
Go for it

**Comment by huytdps13400** [comment]
Implemented in #4037. The fix forwards each custom React element's `accessibilityLabel` and `accessibilityHint` to its ButtonGroup pressable wrapper, matching the VoiceOver/TalkBack use case described here.

Verification: all 44 `@rneui/base` suites pass (339 tests passed, 6 skipped), all 71 snapshots pass, and the package build succeeds.
